lib/bundler/definition.rb

```diff
@@ -94,7 +94,7 @@ def initialize(lockfile, dependencies, sources, unlock, ruby_version = nil, opti
       end
       @unlocking ||= @unlock[:ruby] || (!@locked_ruby_version ^ !@ruby_version)
 
-      @gem_version_promoter = GemVersionPromoter.new(@locked_specs, @unlock[:gems])
+      create_gem_version_promoter
```

Comment:
for testing purposes, maybe make the assignment here instead of in `create_gem_version_promoter` ?